File: javascript/classes/derived.js
class Base {
	constructor(id = 0) {
		this.m_id = id;
	}

	getId() {
		return this.m_id;
	}
};

class Derived extends Base {
	constructor(cost = 0.0) {
		super(id);
		this.m_cost = cost;
	}

	getCost() {
		return this.m_cost;
	}
};

const derived = new Derived(1.3, 5);
console.log(`ID: ${derived.getId()}`);
console.log(`Cost: ${derived.getCost()}`);

File: c++/classes/derived.cpp
#include <iostream>
using namespace std;

class Base
{
public:
	int m_id;

	Base(int id = 0) : m_id(id)
	{}

	int getId() const { return m_id; }
};

class Derived: public Base
{
public:
	double m_cost;

	Derived(double cost = 0.0, int id = 0) : Base(id), m_cost(cost)
	{}

	double getCost() const { return m_cost; }
};

int main()
{
	Derived derived(1.3, 5);
	cout << "ID: " << derived.getId() << '\n';
	cout << "Cost: " << derived.getCost() << '\n';

	return 0;
}
